lib/screens/profile/faq.dart

```diff
@@ -1,110 +1,76 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
-import 'package:auto_route/auto_route.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
+import '../../components/custom_back_button.dart';
+import '../../components/custom_tab_bar/custom_tab_bar.dart';
+import '../../components/status_indicators/error_generic.dart';
+import '../../components/status_indicators/loading_generic.dart';
 import '../../components/web/dialog_on_web.dart';
 import '../../theme/design_system/design_system.dart';
-
+import '../../graphql/queries/faq.graphql.dart';
 import '../../l10n/app_localizations.dart';
 
-class FAQItem {
-  final int id;
-  final String question;
-  final String answer;
-
-  FAQItem({
-    required this.id,
-    required this.question,
-    required this.answer,
-  });
-}
-
-class FAQScreen extends StatefulWidget {
+class FAQScreen extends HookConsumerWidget {
   const FAQScreen({super.key});
 
   @override
-  State<FAQScreen> createState() => _FAQScreenState();
-}
+  Widget build(BuildContext context, WidgetRef ref) {
+    final refreshKey = useState(UniqueKey());
+    final categoriesFuture = useMemoized(
+      () => ref.read(gqlClientProvider).query$FAQ(),
+      [refreshKey.value],
+    );
+    final categoriesSnapshot = useFuture(categoriesFuture);
 
-class _FAQScreenState extends State<FAQScreen> {
-  var listOfList = [
-    FAQItem(
-      id: 1,
-      question: 'Title 1 Title 1 Title 1 Title 1 Title 1 Title1',
-      answer: 'title 2',
-      // question: 'Hvorfor funker dette ikke som jeg forventer?',
-      // answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-    FAQItem(
-      id: 2,
-      question: 'Hvorfor funker dette ikke som jeg forventer?',
-      answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-    FAQItem(
-      id: 3,
-      question: 'Hvorfor funker dette ikke som jeg forventer?',
-      answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-    FAQItem(
-      id: 4,
-      question: 'Hvorfor funker dette ikke som jeg forventer?',
-      answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-    FAQItem(
-      id: 5,
-      question: 'Hvorfor funker dette ikke som jeg forventer?',
-      answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-    FAQItem(
-      id: 6,
-      question: 'Hvorfor funker dette ikke som jeg forventer?',
-      answer: 'Fordi lorem ipsum dolor sit amet consectur osv.',
-    ),
-  ];
+    void refresh() => refreshKey.value = UniqueKey();
+
+    final tempState = useState(0);
+    Future.delayed(const Duration(seconds: 5), () => tempState.value++);
+
+    Widget? child;
+    if (categoriesSnapshot.connectionState == ConnectionState.waiting) {
+      child = const LoadingGeneric();
+    } else if (!categoriesSnapshot.hasData || categoriesSnapshot.data!.parsedData == null) {
```

Comment:
We need to avoid ! where it's unnecessary.
The following is way safer and easier to read:
`else if (categoriesSnapshot.data?.parsedData == null)`

and 

`final categoryItems = categoriesSnapshot.data?.parsedData?.faq.categories?.items;`